source/aggregation/stages.txt

```diff
@@ -0,0 +1,345 @@
+.. _csharp-aggregation-stages:
+
+==================
+Aggregation Stages
+==================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: dotnet, code example, transform, pipeline
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+On this page, you can learn how to create an aggregation pipeline and add stages to it
+by using methods in the {+driver-short+}.
+
+Build an Aggregation Pipeline
+-----------------------------
+
+You can use the {+driver-short+} to build an aggregation pipeline by using builders
+methods or BSON documents. See the following sections to learn more about each of these
+approaches.
+
+.. _csharp-aggregation-stages-builders:
+
+Builders
+~~~~~~~~
+
+You can build a type-safe aggregation pipeline in the following ways:
+
+- Construct an ``EmptyPipelineDefinition<TDocument>`` object. Chain calls from this object
+  to the relevant aggregation methods. Then, pass the pipeline object to the
+  ``IMongoCollection<TDocument>.Aggregate()`` method.
+
+- Call the ``IMongoCollection<TDocument>.Aggregate()`` method. Chain calls from this
+  method call to the relevant aggregation methods.
+
+Select the :guilabel:`EmptyPipelineDefinition`
+or :guilabel:`Aggregate` tab to see the corresponding code for each approach:
+
+.. tabs::
+
+   .. tab:: EmptyPipelineDefinition
+      :tabid: empty-pipeline-definition
+
+      .. code-block:: csharp
+
+         // Defines the aggregation pipeline
+         var pipeline = new EmptyPipelineDefinition<Movie>()
+            .Match(...)
+            .Group(...)
+            .Merge(...);
+
+         // Executes the aggregation pipeline
+         var results = collection.Aggregate(pipeline).ToList();
+
+   .. tab:: Aggregate
+      :tabid: aggregate
+
+      .. code-block:: csharp
+
+         // Defines and executes the aggregation pipeline
+         var results = collection.Aggregate()
+            .Match(...)
+            .Group(...)
+            .Merge(...);
```

Comment:
S: should you also call `ToList()` at the end here?

Reply:
It's probably better to remove the call from the other example.